src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -367,7 +367,7 @@ ShouldHideShardsInternal(void)
 	List *prefixList = NIL;
 
 	/* SplitGUCList scribbles on the input */
-	char *splitCopy = pstrdup(HideShardsFromAppNamePrefixes);
+	char *splitCopy = pstrdup(ShowShardsForAppNamePrefixes);
 
 	if (!SplitGUCList(splitCopy, ',', &prefixList))
 	{
```

Comment:
minor: When invalid `splitCopy`, we show the shards to all apps, should we maybe hide and keep the default behavior same? Otherwise, might be useful to have `DEBUG4` or such when we return here for ease of debugging in the future:) 


```SQL

alter system set citus.show_shards_for_app_name_prefixes to '""';
ALTER SYSTEM
Time: 3.222 ms
select pg_reload_conf();
┌────────────────┐
│ pg_reload_conf │
├────────────────┤
│ t              │
└────────────────┘
(1 row)

Time: 0.375 ms
 \d
                    List of relations
┌────────┬─────────────────────────┬───────┬─────────────┐
│ Schema │          Name           │ Type  │    Owner    │
├────────┼─────────────────────────┼───────┼─────────────┤
│ public │ citus_tables            │ view  │ onderkalaci │
│ public │ pg_stat_statements      │ view  │ onderkalaci │
│ public │ pg_stat_statements_info │ view  │ onderkalaci │
│ public │ test                    │ table │ onderkalaci │
│ public │ test_102008             │ table │ onderkalaci │
│ public │ test_102016             │ table │ onderkalaci │
│ public │ test_102024             │ table │ onderkalaci │
│ public │ test_102032             │ table │ onderkalaci │
└────────┴─────────────────────────┴───────┴─────────────┘
(8 rows)

```

Reply:
This is unreachable due to ShowShardsForAppNamePrefixesCheckHook, but changed anyway.

Setting to empty string means show to any application_name that starts with a prefix of length 0, so that matches everything.